autogalaxy/analysis/clump_model.py

```diff
@@ -52,38 +56,61 @@ def __init__(
 
         self.einstein_radius_upper_limit = einstein_radius_upper_limit
+        self.unfix_centres = unfix_centres
+
 
     @property
     def total_clumps(self) -> int:
         return len(self.centres.in_list)
 
     @property
     def light_list(self) -> Optional[List[af.Model]]:
-        """
-        Returns a list of every clump's light model, where the centre of that light model is fixed to its corresponding
-        input clump's centre.
+        """                                                                                                                                                                               
+        Returns a list of every clump's light model, where the centre of that light model is fixed to its corresponding                                                                   
+        input clump's centre, unless specified to be free.                                                                                                                                
         """
         if self.light_cls is None:
             return None
 
-        return [
-            af.Model(self.light_cls, centre=centre) for centre in self.centres.in_list
-        ]
+        light_list = []
+
+        for centre in self.centres.in_list:
+            if self.unfix_centres:
+                light = af.Model(self.light_cls)
+                light.centre.centre_0 = af.UniformPrior(
+                    lower_limit=centre[0] - 0.1, upper_limit=centre[0] + 0.1
+                )
+                light.centre.centre_1 = af.UniformPrior(
+                    lower_limit=centre[1] - 0.1, upper_limit=centre[1] + 0.1
+                )
+                light_list.append(light)
+            else:
+                light_list.append(af.Model(self.light_cls, centre=centre))
+
+        return light_list
 
     @property
     def mass_list(self) -> Optional[List[af.Model]]:
-        """
-        Returns a list of every clump's mass model, where the centre of that mass model is fixed to its corresponding
-        input clump's centre.
+        """                                                                                                                                                                               
+        Returns a list of every clump's mass model, where the centre of that mass model is fixed to its corresponding                                                                     
+        input clump's centre, unless specified to be free.                                                                                                                                
         """
         if self.mass_cls is None:
             return None
 
         mass_list = []
 
         for centre in self.centres.in_list:
-
-            mass = af.Model(self.mass_cls, centre=centre)
+            if self.unfix_centres:
+                mass = af.Model(self.mass_cls)
+                mass.centre.centre_0 = af.UniformPrior(

```

Comment:
Also consider writing a function 'unfix_centre' to avoid repeating the same code.

```python
mass.centre = unfix_centre(mass.centre)